web/src/screens/private/Schedules/AddSchedule.js

```diff
@@ -0,0 +1,433 @@
+    useDates: false,
+  };
+
+  onGroupChange = (e) => {
+    const groupId = e.target.value;
+    this.setState(() => ({ groupId }));
+  };
+
+  onPriorityChange = (e) => {
+    const priority = e.target.value;
+    this.setState(() => ({ priority }));
+  };
+
+  onCapabilityChange = (e) => {
+    const capability = e.target.value;
+    this.setState(() => ({ capability }));
+  };
+
+  onNameChange = (e) => {
+    const name = e.target.value;
+    this.setState(() => ({ name }));
+  };
+
+  onDetailsChange = (e) => {
+    const details = e.target.value;
+    this.setState(() => ({ details }));
+  };
+
+  onStartTimeChange = (e) => {
+    const startTime = e.target.value;
+    this.setState(() => ({ startTime }));
+  };
+
+  onEndTimeChange = (e) => {
+    const endTime = e.target.value;
+    this.setState(() => ({ endTime }));
+  };
+
+  onUseDatesChange = () => {
+    const useDates = !this.state.useDates;
+    this.setState(() => ({ useDates }));
+  };
+
+  handleNext = () => {
+    const steps = getSteps();
+    if (this.state.activeStep === steps.length - 1) {
+      const { name, details, useDates, startTime, endTime, groupId } = this.state;
+      let startTimeUnix = 0;
+      let endTimeUnix = 2147483647;
+      if (useDates) {
+        startTimeUnix = moment(startTime).unix();
+        endTimeUnix = moment(endTime).unix();
+      }
+      this.props
+        .createSchedule({ name, details, startTime: startTimeUnix, endTime: endTimeUnix, groupId })
+        .then(() => {
+          const { history } = this.props;
+          history.push('/dashboard');
+        })
+        .catch((error) => {
+          this.setState(() => ({ message: error.message, open: true }));
+          setTimeout(() => {
+            this.setState(() => ({ message: '', open: false }));
+          }, 3000);
+        });
+    } else {
+      this.setState({
+        activeStep: this.state.activeStep + 1,
+      });
+    }
+  };
+
+  handleBack = () => {
+    this.setState({
+      activeStep: this.state.activeStep - 1,
+    });
+  };
+
+  handleReset = () => {
+    this.setState({
+      activeStep: 0,
+    });
+  };
+
+  render() {
+    const { classes } = this.props;
+    const steps = getSteps();
+    const { activeStep } = this.state;
+
+    if (this.props.loading) {
+      return <CircularProgress className={classes.progress} size={50} />;
+    }
+    return (
+      <form className={classes.root}>
+        <Card className={classes.card}>
+          <CardContent className={classes.cardContent}>
+            <div style={{ display: 'flex' }}>
+              <NavLink to="/dashboard">
+                <ArrowBackIcon className={classes.cardIcon} />
+              </NavLink>
+              <Typography variant="title" color="inherit" className={classes.cardTitle}>
+                Add New Request
+              </Typography>
+            </div>
+            <Stepper activeStep={activeStep} orientation="vertical">
+              {steps.map((label, index) => (
```

Comment:
not important: I feel like putting these in separate components or something might not be a terrible idea. This function/JSX block is massive.

Reply:
Yes I agree - just worked on getting it worker and then I'll split what I can into smaller components.